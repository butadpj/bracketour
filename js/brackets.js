import { names } from '../api/names.js';
import {
  getRandomIndexFrom,
  groupArrayBy,
  randomNumberBetween,
} from './utils/index.js';

/* 
  TODO: FIX BUG WHEN RANDOM SCORE IS STILL GENERATING 
  EVEN WHEN THERE'S ONLY ONE (1) PLAYER LEFT
*/

// SELECTORS
const participantForm = document.getElementById('participant-form');
const brackets = document.getElementById('brackets');

const editBracketButton = document.getElementById('edit-btn');
const nextRoundButton = document.getElementById('next-round');
//

// GLOBALS
let nextRoundButtonState = 1;
let playersGroupCount;
//

// FUNCTIONS
const getAllRoundLines = (maxRound = 7) => {
  let roundLines = [];
  for (let i = 1; i <= maxRound; i++) {
    roundLines.push(document.getElementById(`lines-r${i}`));
  }
  return roundLines;
};

const getAllRoundConnectors = (maxRound = 7) => {
  let roundConnectors = [];
  for (let i = 1; i <= maxRound; i++) {
    roundConnectors.push(document.getElementById(`connectors-r${i}`));
  }
  return roundConnectors;
};

const getAllRoundPlayers = (maxRound = 7) => {
  let roundPlayers = [];
  for (let i = 1; i <= maxRound; i++) {
    roundPlayers.push(document.getElementById(`players-r${i}`));
  }
  return roundPlayers;
};

const selectRoundFromAll = (round, getter) => {
  if (round === 0) return console.log('Round starts from 1');
  return getter()[round - 1];
};

const showBrackets = (participantCount) => {
  participantForm.classList.add('hidden');
  editBracketButton.classList.remove('hidden');
  brackets.classList.remove('hidden');
  nextRoundButton.classList.remove('hidden');

  createBracketsFromRandomNames(participantCount);
};

const createBracketsFromRandomNames = (playersCount) => {
  playersGroupCount = Math.ceil(playersCount / 2);
  populateContainerWithRandomPlayers(
    playersCount,
    selectRoundFromAll(1, getAllRoundPlayers),
  );
};

const getUniqueNames = (count) => {
  const uniqueNames = new Set();
  let randomName;

  // Getting unique names
  do {
    randomName = names[getRandomIndexFrom(names)];
    uniqueNames.add(randomName);
  } while (uniqueNames.has(randomName) && uniqueNames.size < count);

  return uniqueNames;
}

const populateContainerWithRandomPlayers = (count, container) => {
  const uniqueNames = getUniqueNames(count);
  
  // Using unique names
  uniqueNames.forEach((value) => {
    container.appendChild(createPlayerElement({ name: value, score: 0 }));
  });
};

const populateContainerWithConnectors = (count, container) => {
  for (let i = 0; i < count; i++) {
    const connectorElement = document.createElement('div');
    container.appendChild(connectorElement);
  }
};

const populateContainerWithLines = (count, container) => {
  for (let i = 0; i < count; i++) {
    const lineElement = document.createElement('div');
    container.appendChild(lineElement);
  }
};

const createPlayerElement = ({ name, score }) => {
  const player = document.createElement('div');
  player.classList.add('player');

  player.innerHTML = `
    <div class="player__name" data-name="${name}">${name}</div>
    <div class="player__score">${score}</div>
  `;

  return player;
};

const getWinnersFromGroupedArray = (groupedArray) => {
  let winners = [];

  const selectRandomWinner = (array) => {
    if (array.length > 1) array[randomNumberBetween(0, 1)];
    return array[0];
  };

  groupedArray.forEach((group) => {
    let winner = selectRandomWinner(group);
    winner.querySelector('.player__score').textContent = randomNumberBetween(
      2,
      3,
    );
    winners.push(winner);

    // Loop through [div.player, div.player]
    // The element that still has score of 0 is the loser
    group.forEach((element) => {
      const score = element.querySelector('.player__score').textContent;
      if (score == 0) {
        const loser = element;
        loser.querySelector('.player__score').textContent = randomNumberBetween(
          0,
          1,
        );
      }
    });
  });

  return winners;
};

const getRoundWinners = (roundPlayers, groupCount) => {
  const groupedPlayers = groupArrayBy([...roundPlayers.children], groupCount);
  const roundWinners = getWinnersFromGroupedArray(groupedPlayers);

  return roundWinners;
};

const getFinalWinner = (roundPlayers) => {
  if (roundPlayers.childElementCount === 1) {
    const finalWinner = roundPlayers.firstElementChild;
    const winnerName = finalWinner.querySelector('.player__name').textContent;

    finalWinner.classList.add('winner');
    finalWinner.querySelector('.player__name').classList.add('winner');

    finalWinner.querySelector('.player__score').textContent = 'W';

    nextRoundButton.disabled = true;
    nextRoundButton.classList.add('disabled');

    setTimeout(() => {
      alert(`The winner of this tournament is: ${winnerName}`);
    }, 200);
  }
  return
};

const advanceWinnersToNextRound = (
  roundWinners,
  insertNextToThis,
  container,
) => {
  // Append winners to the container
  roundWinners.forEach((winner) => {
    // Appending element to somewhere will move the element
    // Cloning it first, and then moving the cloned element
    // solves the issue
    const clonedWinner = winner.cloneNode(true);
    clonedWinner.querySelector('.player__score').textContent = 0;
    container.appendChild(clonedWinner);
  });

  // Insert container with winners next to specified element
  insertNextToThis.insertAdjacentElement('afterend', container);
};

const handleNextRoundButton = (currentRound) => {
  const roundWinners = getRoundWinners(
    selectRoundFromAll(currentRound, getAllRoundPlayers),
    playersGroupCount,
  );

  advanceWinnersToNextRound(
    roundWinners,
    selectRoundFromAll(currentRound, getAllRoundLines),
    selectRoundFromAll(currentRound + 1, getAllRoundPlayers),
  );

  populateContainerWithConnectors(
    selectRoundFromAll(currentRound, getAllRoundPlayers).childElementCount / 2,
    selectRoundFromAll(currentRound, getAllRoundConnectors),
  );
  
  populateContainerWithLines(
    selectRoundFromAll(currentRound, getAllRoundPlayers).childElementCount / 4,
    selectRoundFromAll(currentRound, getAllRoundLines),
  );

  // Reduce the player groupings by half
  playersGroupCount = Math.ceil(playersGroupCount / 2);

  getFinalWinner(selectRoundFromAll(currentRound + 1, getAllRoundPlayers));
} 
//

// DOM nodes needed to be loaded first
// For testing purposes
window.addEventListener('load', () => {
  // EVENT LISTENERS

  // hover feature start
  brackets.addEventListener('mouseover', (e) => {
    if (e.target.classList.contains('player')) {
      const name = e.target.firstElementChild.dataset.name;
      const mathcedPlayers = document.querySelectorAll(`[data-name='${name}']`);

      mathcedPlayers.forEach((player) => {
        player.closest('.player').classList.add('active');
        player.classList.add('active');
      });
    }

    if (
      e.target.classList.contains('player__name') ||
      e.target.classList.contains('player__score')
    ) {
      const name = e.target.closest('.player').firstElementChild.dataset.name;
      const mathcedPlayers = document.querySelectorAll(`[data-name='${name}']`);

      mathcedPlayers.forEach((player) => {
        player.closest('.player').classList.add('active');
        player.classList.add('active');
      });
    }
  });

  brackets.addEventListener('mouseout', (e) => {
    if (e.target.classList.contains('player')) {
      const name = e.target.firstElementChild.dataset.name;
      const mathcedPlayers = document.querySelectorAll(`[data-name='${name}']`);

      mathcedPlayers.forEach((player) => {
        player.closest('.player').classList.remove('active');
        player.classList.remove('active');
      });
    }

    if (
      e.target.classList.contains('player__name') ||
      e.target.classList.contains('player__score')
    ) {
      const name = e.target.closest('.player').firstElementChild.dataset.name;
      const mathcedPlayers = document.querySelectorAll(`[data-name='${name}']`);

      mathcedPlayers.forEach((player) => {
        player.closest('.player').classList.remove('active');
        player.classList.remove('active');
      });
    }
  });
  // hover feature end

  // edit bracket feature start
  editBracketButton.addEventListener('click', () => {
    participantForm.classList.remove('hidden');
    editBracketButton.classList.add('hidden');
    brackets.classList.add('hidden');
    nextRoundButton.classList.add('hidden');

    getAllRoundPlayers().forEach((roundPlayers) => {
      roundPlayers.innerHTML = '';
    });

    getAllRoundConnectors().forEach((roundConnectors) => {
      roundConnectors.innerHTML = '';
    });

    getAllRoundLines().forEach((roundLines) => {
      roundLines.innerHTML = '';
    });

    nextRoundButtonState = 1;
    nextRoundButton.disabled = false;
    nextRoundButton.classList.remove('disabled');
  });
  // edit bracket feature end

  // next round feature start
  nextRoundButton.addEventListener('click', () => {
    handleNextRoundButton(nextRoundButtonState);
    nextRoundButtonState++;
  });
  // next round feature end
});

// EXPOSE
export { 
  showBrackets, 
  createPlayerElement,  
  getUniqueNames,
};
//
