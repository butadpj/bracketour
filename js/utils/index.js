export { addClassTo } from './addClassTo.js';
export { getRandomIndexFrom } from './getRandomIndexFrom.js';
export { regexValidate } from './regexValidate.js';
export { isOdd } from './isOdd.js';
export { randomNumberBetween } from './randomNumberBetween.js';
export { groupArrayBy } from './groupArrayBy.js';
