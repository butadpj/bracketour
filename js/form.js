import { showBrackets } from './brackets.js';

const participantForm = document.getElementById('participant-form');

participantForm.addEventListener('submit', (e) => {
  e.preventDefault();
  let participantCount = e.target.querySelector('#participant-input').value;
  showBrackets(participantCount);
})

