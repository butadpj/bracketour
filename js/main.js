import { regexValidate, addClassTo } from './utils/index.js';

const participantInput = document.getElementById('participant-input');

const minElement = document.querySelector('.constraints .min');
const maxElement = document.querySelector('.constraints .max'); 

const submitBtn = document.getElementById('submit-btn');

const numbersOnly = regexValidate('number');

participantInput.addEventListener('input', (e) => {
  const input = e.target.value;

  const maxPraticipants = 50;
  const minPraticipants = 2;

  const validateInput = numbersOnly(pattern => {
    if (
        input && // not empty
        pattern.test(input) && // valid number
        Number(input) <= maxPraticipants &&
        Number(input) >= minPraticipants
      ) { 
        return {
          valid: true,
          value: input
        }
    } 

    if (Number(input) > maxPraticipants) return {
      valid: false,
      error: 'max'
    }

    if (Number(input) < minPraticipants) return {
      valid: false,
      error: 'min'
    }

    // default
    return {
      valid: false,
      error: 'Please input a valid number',
    }
  });

  if (!validateInput.valid) {
    participantInput.classList.add('danger');
    participantInput.classList.remove('success');
    submitBtn.classList.add('disabled');

    submitBtn.disabled = true;

    if (validateInput.error === 'min') {
      const removeAfterSecond = addClassTo('apply-shake', minElement);
      removeAfterSecond(1);
    }
    if (validateInput.error === 'max') {
      const removeAfterSecond = addClassTo('apply-shake', maxElement);
      removeAfterSecond(1);
    }
    return
  } 

  participantInput.classList.add('success');
  participantInput.classList.remove('danger');  
  submitBtn.classList.remove('disabled');

  submitBtn.disabled = false;
});

