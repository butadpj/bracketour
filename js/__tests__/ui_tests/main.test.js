import $ from 'jquery';
import {
  numberPattern,
  validateInputWithPattern,
  validInputAndButton,
  invalidInputAndButton,
  MAX_PARTICIPANTS,
  MIN_PARTICIPANTS,
} from '../../main.js';

import { addClassTo } from '../../utils/addClassTo.js';

describe('main.js', () => {
  document.body.innerHTML = `
      <input
        id="participant-input"
        class="form__input"
        type="text"
        placeholder="2"
      />
      <div class="constraints">
        <h4 class="min">min: <span>2</span></h4>
        <h4 class="max">max: <span>50</span></h4>
      </div>
      <button
        disabled
        id="submit-btn"
        type="submit"
        class="btn submit__btn disabled"
      >
        Generate
      </button>
    `;

  const participantInput = $('#participant-input')[0];

  const minConstraintElement = $('.min')[0];
  const maxConstraintElement = $('.min')[0];

  const minConstraintValue = $('.min span')[0];
  const maxConstraintValue = $('.max span')[0];

  const submitBtn = $('#submit-btn')[0];

  const testInput = (input) => {
    participantInput.value = input;

    const inputResult = validateInputWithPattern(
      participantInput.value,
      numberPattern,
    );

    if (!inputResult.valid) {
      invalidInputAndButton(participantInput, submitBtn);
    } else {
      validInputAndButton(participantInput, submitBtn);
    }

    return inputResult;
  };

  jest.useFakeTimers();
  jest.spyOn(global, 'setTimeout');

  it('add class "danger" to input if the input is invalid', () => {
    testInput(100);
    expect(document.body).toMatchSnapshot();
    expect(participantInput.classList.contains('danger')).toBe(true);

    testInput(1);
    expect(document.body).toMatchSnapshot();
    expect(participantInput.classList.contains('danger')).toBe(true);

    testInput(2);
    expect(document.body).toMatchSnapshot();
    expect(participantInput.classList.contains('danger')).toBe(false);

    testInput(30);
    expect(document.body).toMatchSnapshot();
    expect(participantInput.classList.contains('danger')).toBe(false);
  });

  it('add class "success" to input if the input is valid', () => {
    testInput(300);
    expect(document.body).toMatchSnapshot();
    expect(participantInput.classList.contains('success')).toBe(false);

    testInput(-5);
    expect(document.body).toMatchSnapshot();
    expect(participantInput.classList.contains('success')).toBe(false);

    testInput(2);
    expect(document.body).toMatchSnapshot();
    expect(participantInput.classList.contains('success')).toBe(true);

    testInput(50);
    expect(document.body).toMatchSnapshot();
    expect(participantInput.classList.contains('success')).toBe(true);
  });

  it('min/max constraint values are matched to MIN/MAX constants', () => {
    expect(minConstraintValue.textContent).toBe(String(MIN_PARTICIPANTS));
    expect(maxConstraintValue.textContent).toBe(String(MAX_PARTICIPANTS));
  });

  it(`
      add class "apply-shake" to min constriant 
      element if input is below MIN constant, and
      remove it after n second(s)
    `, () => {
    const inputResult1 = testInput(1);

    if (inputResult1.error === 'min') {
      const removeAfterSecond = addClassTo('apply-shake', minConstraintElement);
      removeAfterSecond(1);
    }

    expect(minConstraintElement.classList.contains('apply-shake')).toBe(true);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000);

    const inputResult2 = testInput(-5);

    if (inputResult2.error === 'min') {
      const removeAfterSecond = addClassTo('apply-shake', minConstraintElement);
      removeAfterSecond(5);
    }
    expect(minConstraintElement.classList.contains('apply-shake')).toBe(true);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 5000);

    expect(document.body).toMatchSnapshot();
  });

  it(`
      add class "apply-shake" to max constriant 
      element if input is higher than MAX constant, and
      remove it after n second(s)
    `, () => {
    const inputResult1 = testInput(69);

    if (inputResult1.error === 'max') {
      const removeAfterSecond = addClassTo('apply-shake', maxConstraintElement);
      removeAfterSecond(1);
    }
    expect(maxConstraintElement.classList.contains('apply-shake')).toBe(true);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000);

    const inputResult2 = testInput(1000);

    if (inputResult2.error === 'max') {
      const removeAfterSecond = addClassTo('apply-shake', maxConstraintElement);
      removeAfterSecond(3);
    }
    expect(maxConstraintElement.classList.contains('apply-shake')).toBe(true);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 3000);

    expect(document.body).toMatchSnapshot();
  });

  it('button is disabled unless the input is valid', () => {
    testInput(100);
    expect(document.body).toMatchSnapshot();
    expect(submitBtn.disabled).toBe(true);

    testInput(1);
    expect(document.body).toMatchSnapshot();
    expect(submitBtn.disabled).toBe(true);

    testInput(2);
    expect(document.body).toMatchSnapshot();
    expect(submitBtn.disabled).toBe(false);

    testInput(30);
    expect(document.body).toMatchSnapshot();
    expect(submitBtn.disabled).toBe(false);
  });
});
