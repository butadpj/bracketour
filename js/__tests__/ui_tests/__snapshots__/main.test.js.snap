// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main.js 
      add class "apply-shake" to max constriant 
      element if input is higher than MAX constant, and
      remove it after n second(s)
     1`] = `
<body>
  
      
  <input
    class="form__input danger"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min apply-shake"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn disabled"
    disabled=""
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js 
      add class "apply-shake" to min constriant 
      element if input is below MIN constant, and
      remove it after n second(s)
     1`] = `
<body>
  
      
  <input
    class="form__input danger"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min apply-shake"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn disabled"
    disabled=""
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js add class "danger" to input if the input is invalid 1`] = `
<body>
  
      
  <input
    class="form__input danger"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn disabled"
    disabled=""
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js add class "danger" to input if the input is invalid 2`] = `
<body>
  
      
  <input
    class="form__input danger"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn disabled"
    disabled=""
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js add class "danger" to input if the input is invalid 3`] = `
<body>
  
      
  <input
    class="form__input success"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn"
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js add class "danger" to input if the input is invalid 4`] = `
<body>
  
      
  <input
    class="form__input success"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn"
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js add class "success" to input if the input is valid 1`] = `
<body>
  
      
  <input
    class="form__input danger"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn disabled"
    disabled=""
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js add class "success" to input if the input is valid 2`] = `
<body>
  
      
  <input
    class="form__input danger"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn disabled"
    disabled=""
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js add class "success" to input if the input is valid 3`] = `
<body>
  
      
  <input
    class="form__input success"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn"
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js add class "success" to input if the input is valid 4`] = `
<body>
  
      
  <input
    class="form__input success"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn"
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js button is disabled unless the input is valid 1`] = `
<body>
  
      
  <input
    class="form__input danger"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min apply-shake"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn disabled"
    disabled=""
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js button is disabled unless the input is valid 2`] = `
<body>
  
      
  <input
    class="form__input danger"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min apply-shake"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn disabled"
    disabled=""
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js button is disabled unless the input is valid 3`] = `
<body>
  
      
  <input
    class="form__input success"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min apply-shake"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn"
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;

exports[`main.js button is disabled unless the input is valid 4`] = `
<body>
  
      
  <input
    class="form__input success"
    id="participant-input"
    placeholder="2"
    type="text"
  />
  
      
  <div
    class="constraints"
  >
    
        
    <h4
      class="min apply-shake"
    >
      min: 
      <span>
        2
      </span>
    </h4>
    
        
    <h4
      class="max"
    >
      max: 
      <span>
        50
      </span>
    </h4>
    
      
  </div>
  
      
  <button
    class="btn submit__btn"
    id="submit-btn"
    type="submit"
  >
    
        Generate
      
  </button>
  
    
</body>
`;
